databases/week3/meal-sharing.sql

```diff
@@ -0,0 +1,348 @@
+) 
+VALUES 
+  (
+    5, 'Nice', 'Liked it!', 2, 4, '2021-12-02'
+  );
+INSERT INTO review (
+  `id`, `title`, `description`, `meal_id`, 
+  `stars`, `created_date`
+) 
+VALUES 
+  (
+    6, 'Perfect', 'Best sandwich ever', 
+    2, 5, '2021-12-02'
+  );
+
+-- Task queries
+-- Get all meals
+SELECT 
+  * 
+FROM 
+  meal;
+
+-- Add a new meal
+INSERT INTO meal (
+  `id`, `title`, `description`, `location`, 
+  `when`, `max_reservations`, `price`, 
+  `created_date`
+) 
+VALUES 
+  (
+    4, 'Home-made sushi', 'Vegan variations included', 
+    'Copenhagen', '2022-04-08 18:00:00', 
+    4, 80, '2022-03-29'
+  );
+
+-- Get a meal with any id, fx 1
+SELECT 
+  * 
+FROM 
+  meal 
+WHERE 
+  id = 2;
+
+-- Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE 
+  meal 
+SET 
+  max_reservations = 5 
+WHERE 
+  id = 2;
+
+-- Delete a meal with any id, fx 1
+DELETE FROM 
+  meal 
+WHERE 
+  title = 'Currywurst';
+
+-- Reservation
+-- Get all reservations
+SELECT 
+  * 
+FROM 
+  reservation;
+
+-- Add a new reservation
+INSERT INTO reservation (
+  `id`, `number_of_guests`, `meal_id`, 
+  `created_date`, `contact_phonenumber`, 
+  `contact_name`, `contact_email`
+) 
+VALUES 
+  (
+    4, 4, 4, '2022-03-30', '71624115', 'Joe', 
+    'joe@gmail.com'
+  );
+
+-- Get a reservation with any id, fx 1
+SELECT 
+  * 
+FROM 
+  reservation 
+WHERE 
+  id = 3;
+
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE 
+  reservation 
+SET 
+  `contact_phonenumber` = '32465786' 
+WHERE 
+  id = 3;
+
+-- Delete a reservation with any id, fx 1
+DELETE FROM 
+  reservation 
+WHERE 
+  id = 1;
+-- Get all reviews
+SELECT 
+  * 
+FROM 
+  review;
+
+-- Add a new review
+INSERT INTO review (
+  `id`, `title`, `description`, `meal_id`, 
+  `stars`, `created_date`
+) 
+VALUES 
+  (
+    3, 'Super tasty', 'Exactly like it should be', 
+    1, 5, '2021-12-02'
+  );
+INSERT INTO review (
+  `id`, `title`, `description`, `meal_id`, 
+  `stars`, `created_date`
+) 
+VALUES 
+  (
+    7, 'Nice', 'The fish was a bit too salty though', 
+    4, 4, '2021-12-02'
+  );
+
+-- Get a review with any id, fx 1
+SELECT 
+  * 
+FROM 
+  review 
+WHERE 
+  id = 1;
+
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE 
+  review 
+SET 
+  meal_id = 2 
+WHERE 
+  id = 3;
+
+-- Delete review
+DELETE FROM 
+  review 
+WHERE 
+  id = 1;
+
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT 
+  * 
+FROM 
+  meal 
+WHERE 
+  price < 50;
+
+-- Get meals that still has available reservations
+SELECT 
+  meal.id, 
+  meal.title, 
+  meal.description, 
+  meal.location, 
+  meal.price, 
+  meal.max_reservations - SUM(reservation.number_of_guests) AS `reservations_available` 
+FROM 
+  meal 
+  JOIN reservation ON meal.id = reservation.meal_id 
+GROUP BY 
+  meal.id 
+ORDER BY 
+  `reservations_available` DESC;
```

Comment:
Get meals that `STILL`has available reservations.
What exactly does this give out..? This returns all meals no matter how many reservations they have available(left).
As meal id=1 is fully booked it shouldn't be included here. We are only looking for meals still have reservation spots left.
Think about it and let me know if you need further hints....
<img width="605" alt="image" src="https://user-images.githubusercontent.com/61273043/160026046-9e69e5d2-2d30-47a3-9ada-5bbad3468b33.png">

SELECT 
  meal.id, 
  meal.title, 
  meal.description, 
  meal.location, 
  meal.price, 
  meal.max_reservations,
  meal.max_reservations - SUM(reservation.number_of_guests) AS `reservations_available` 
FROM 
  meal 
  JOIN reservation ON meal.id = reservation.meal_id 
GROUP BY 
  meal.id 
ORDER BY 
  `reservations_available` DESC;
```

Reply:
fixed :)